core/node/fee_model/src/l1_gas_price/gas_adjuster/mod.rs

```diff
@@ -197,36 +329,33 @@ impl GasAdjuster {
                     * BLOB_GAS_PER_BYTE as f64
                     * self.config.internal_pubdata_pricing_multiplier;
 
-                self.bound_blob_base_fee(calculated_price)
-            }
-            PubdataSendingMode::Calldata => {
-                self.estimate_effective_gas_price() * self.pubdata_byte_gas()
+                self.bound_pubdata_fee(calculated_price)
             }
+            PubdataSendingMode::Calldata => self.bound_pubdata_fee(
+                (self.estimate_effective_gas_price() * L1_GAS_PER_PUBDATA_BYTE as u64) as f64,
```

Comment:
Dumb question: Wouldn't it make more sense to convert `estimate_effective_gas_price()` output to `f64` so that there's less chance of integer overflow?

Reply:
The values will be converted to integers in the end anyway, so if it goes above u64::max, there will be issues anyway